test/e2e/case18_discovery_refresh_test.go

```diff
@@ -54,12 +56,33 @@ var _ = Describe("Test discovery info refresh", func() {
 	})
 
 	It("Verifies that the discovery info is refreshed when a template references a new object kind", func() {
+		By("Adding a non-functional API service")
+		utils.Kubectl("apply", "-f", badAPIServiceYaml)
+
+		By("Checking that the API causes an error during API discovery")
+		Eventually(
+			func() interface{} {
+				cmd := exec.Command("kubectl", "api-resources")
+
+				err := cmd.Start()
+				if err != nil {
+					return err
```

Comment:
This error is not indicative of the error you are looking for so `Eventually` shouldn't stop from this error.

Reply:
Nice catch.